core/src/main/kotlin/fr/sncf/osrd/graph/Interfaces.kt

```diff
@@ -4,6 +4,14 @@ import fr.sncf.osrd.utils.indexing.StaticIdx
 import fr.sncf.osrd.utils.units.Length
 import fr.sncf.osrd.utils.units.Offset
 
+/**
+ * This interface defines a function that can be used as a heuristic for an A* pathfinding. It takes
+ * a node as input, and returns an estimation of the remaining distance.
+ */
+fun interface STDCMAStarHeuristic<NodeT> {
+    fun apply(node: NodeT): Double
+}
```

Comment:
We don't necessarily need an abstract interface anymore

We can also change it to be closer to our needs here. Specifically we could have the numbers of passed waypoints as a parameter, instead of having an awkward list of lambdas